src/main/scala/io/qbeast/spark/delta/writer/Compactor.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ */
+package io.qbeast.spark.delta.writer
+
+import io.qbeast.IISeq
+import io.qbeast.core.model.{CubeId, QTableID, QbeastBlock, TableChanges}
+import io.qbeast.spark.utils.{State, TagUtils}
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.mapred.{JobConf, TaskAttemptContextImpl, TaskAttemptID}
+import org.apache.hadoop.mapreduce.TaskType
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.delta.actions.{AddFile, FileAction, RemoveFile}
+import org.apache.spark.sql.execution.datasources.{OutputWriter, OutputWriterFactory}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.util.SerializableConfiguration
+
+import java.util.UUID
+
+/**
+ * Compacts the information from a set of CubeBlocks into a single block
+ * @param tableID the table identifier
+ * @param factory output writer factory
+ * @param serConf configuration to serialize the data
+ * @param schema the schema
+ * @param cubeId the cubeId to compact
+ * @param cubeBlocks the blocks to compact
+ * @param tableChanges the TableChanges
+ */
+
+case class Compactor(
+    tableID: QTableID,
+    factory: OutputWriterFactory,
+    serConf: SerializableConfiguration,
+    schema: StructType,
+    cubeId: CubeId,
+    cubeBlocks: IISeq[QbeastBlock],
+    tableChanges: TableChanges)
+    extends Serializable {
+
+  def writeBlock(it: Iterator[InternalRow]): Iterator[FileAction] = {
+
+    val minWeight = cubeBlocks.map(_.minWeight).min
+    val maxWeight =
```

Comment:
It is not critical, but I wonder if both min and max can be computed traversing the sequence just once.

Reply:
It could be possible. Solved on 3d7ad28